hack/verify-typecheck-providerless.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+cd "${KUBE_ROOT}"
+# verify the providerless build
+# https://github.com/kubernetes/enhancements/blob/master/keps/sig-cloud-provider/20190729-building-without-in-tree-providers.md
+hack/verify-typecheck.sh --skip-test --tags=providerless --ignore-dirs=test
```

Comment:
Did we end up having to skip all test files for now? I thought the test/e2e dir was the main thing we needed to skip

Reply:
yeah, there's also some integration code IIRC...
we have no plan of record for removing these from test yet.
IMO it needs to happen, but I'm not signing myself up just yet, and getting the core binaries to work without them is a start.

we still don't have these things fully removed from the core binaries as well :( (credential provider...)

Reply:
sure, was just curious